src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -62,9 +62,20 @@ public WebScriptHostManager(ScriptHostConfiguration config, ISecretManagerFactor
                 config.TraceWriter = systemTraceWriter;
             }
 
-            _secretManager = secretManagerFactory.Create(settingsManager, config.TraceWriter, new FileSystemSecretsRepository(webHostSettings.SecretsPath));
             _performanceManager = new HostPerformanceManager(settingsManager);
             _swaggerDocumentManager = new SwaggerDocumentManager(config);
+            
+            var secretStorageType = settingsManager.GetSetting("UseBlobSecretStorage"); //TODO: official parameter naming                        
+            if (secretStorageType != null) 
```

Comment:
Indeed, as we've discussed, we want to introduce a factory and deal with the creation, as well as the logic to decide which repository needs to be created.